### PR TITLE
Update InferenceSession call to latest format

### LINKED ISSUE #7980: ONNX tests fail due to upstream bug
For example https://github.com/pytorch/vision/actions/runs/6259955045?pr=7979#summary-16997013188

This was introduced in the `1.16.0` release that was published yesterday. See microsoft/onnxruntime#17631

cc @neginraoof @justinchuby @seemethere

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Fixes: #7980 
Reverts: #7981

The actual fix is a simple update call convention for onnxruntime. No need to skip latest ORT version

@vfdev-5 @seemethere @pmeier 

cc @pmeier @neginraoof @justinchuby

### CODE REVIEW

File: test/test_onnx.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def to_numpy(tensor):
         inputs = list(map(to_numpy, inputs))
         outputs = list(map(to_numpy, outputs))
 
-        ort_session = onnxruntime.InferenceSession(onnx_io.getvalue())
+        ort_session = onnxruntime.InferenceSession(onnx_io.getvalue(),
+                                                   providers=onnxruntime.get_available_providers())
```
Comment (pmeier) [inline test/test_onnx.py:83]: Lint is failing: https://github.com/pytorch/vision/actions/runs/6271822144/job/17032160789?pr=7982#step:11:410

```suggestion
        ort_session = onnxruntime.InferenceSession(onnx_io.getvalue(), providers=onnxruntime.get_available_providers())
```

**Comment by pmeier** [comment]
ONNX job is green: https://github.com/pytorch/vision/actions/runs/6271822156/job/17032158532?pr=7982

@justinchuby can you review this PR?

**Comment by thiagocrepaldi** [comment]
> ONNX job is green: https://github.com/pytorch/vision/actions/runs/6271822156/job/17032158532?pr=7982
> 
> @justinchuby can you review this PR?

We need reviews from @vfdev-5 @seemethere for merging :)

**Comment by pmeier** [comment]
> We need reviews from @vfdev-5 @seemethere for merging :)

We don't. My approval is enough. But since I don't have much ONNX experience, I wanted to make sure the fix is good.

**Review by pmeier — APPROVED** [review]
Thanks @thiagocrepaldi!

**Pull request merged**